gameobjects/ball.cpp

```diff
@@ -20,13 +20,22 @@ static constexpr float Y_STARTING_POSITION_BOUNDS = 500.0f;
 
 static constexpr std::chrono::seconds BALL_RESET_WAIT_S { 3 };
 
+void Ball::InitalizeComponents()
+{
+    AddComponent<Transform>(glm::vec3(0.0f));
+    AddComponent<Rectangle>(BALL_WIDTH, BALL_WIDTH);
+    AddComponent<ColliderBox>(BALL_WIDTH, BALL_WIDTH);
```

Comment:
Cool. I really like this for a lot of reasons.

You see how this approach naturally separates concerns and interfaces into individual sub classes. Now when you want to change something that "is transformable" you know where to look. You don't have a base class doing it, you don't have to update it in a billion subclasses.


This is really mature architecture, very thought provoking.

Reply:
Yeah it is definitely a lot nicer 🎉